server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTaskCopied.java

```diff
@@ -0,0 +1,329 @@
+import com.oceanbase.odc.service.task.TaskService;
+import com.oceanbase.odc.service.task.constants.JobParametersKeyConstants;
+import com.oceanbase.odc.service.task.enums.JobStatus;
+import com.oceanbase.odc.service.task.model.ExecutorInfo;
+import com.oceanbase.odc.service.task.runtime.PreCheckTask;
+import com.oceanbase.odc.service.task.runtime.PreCheckTaskParameters;
+import com.oceanbase.odc.service.task.schedule.DefaultJobDefinition;
+import com.oceanbase.odc.service.task.schedule.JobDefinition;
+import com.oceanbase.odc.service.task.schedule.JobScheduler;
+import com.oceanbase.odc.service.task.service.TaskFrameworkService;
+import com.oceanbase.odc.service.task.util.JobUtils;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author gaoda.xy
+ * @date 2024/1/30 11:06
+ */
+@Slf4j
+public class PreCheckRuntimeFlowableTaskCopied extends BaseODCFlowTaskDelegate<Void> {
+
+    @Autowired
+    private PreCheckTaskProperties preCheckTaskProperties;
+    @Autowired
+    private EnvironmentService environmentService;
+    @Autowired
+    private RuleService ruleService;
+    @Autowired
+    private DatabaseService databaseService;
+    @Autowired
+    private ObjectStorageFacade objectStorageFacade;
+    @Autowired
+    private ApprovalFlowConfigSelector approvalFlowConfigSelector;
+    @Autowired
+    private TaskFrameworkService taskFrameworkService;
+    @Autowired
+    private JobScheduler jobScheduler;
+
+    private Long preCheckTaskId;
+    private PreCheckTaskResult preCheckResult;
+    private volatile boolean success = false;
+
+    private static final String CHECK_RESULT_FILE_NAME = "sql-check-result.json";
+
+    @Override
+    protected Void start(Long taskId, TaskService taskService, DelegateExecution execution) throws Exception {
+        serviceTaskRepository.updateStatusById(getTargetTaskInstanceId(), FlowNodeStatus.EXECUTING);
+        this.preCheckTaskId = FlowTaskUtil.getPreCheckTaskId(execution);
+        RiskLevelDescriber riskLevelDescriber = FlowTaskUtil.getRiskLevelDescriber(execution);
+        TaskEntity taskEntity;
+        ConnectionConfig connectionConfig = null;
+        try {
+            taskEntity = taskService.detail(taskId);
+        } catch (Exception e) {
+            throw new ServiceTaskError(e);
+        }
+        try {
+            connectionConfig = FlowTaskUtil.getConnectionConfig(execution);
+        } catch (VerifyException e) {
+            log.info(e.getMessage());
+        }
+        if (Objects.nonNull(connectionConfig) && needCreatePreCheckTask(taskEntity)) {
+            // Skip SQL pre-check if connection config is null
+            JobDefinition jobDefinition = buildJobDefinition(taskEntity, connectionConfig, riskLevelDescriber);
+            Long jobId = jobScheduler.scheduleJobNow(jobDefinition);
+            taskService.updateJobId(this.preCheckTaskId, jobId);
+            try {
+                // Pre-check task has no timeout yet, set to max value (about 24 days)
```

Comment:
timeout is to lang, may 1 hours

Reply:
Yes you are right, I add a system config to describe it and the default vulue is 1 hour.